addons/hr_expense/models/product_product.py

```diff
@@ -24,3 +24,33 @@ def _compute_standard_price_update_warning(self):
                             "There are unsubmitted expenses linked to this category. Updating the category cost will change expense amounts. "
                             "Make sure it is what you want to do."
                         )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'standard_price' not in vals:  # Only check for expenses if the standard price has been updated
+            return res
+
+        expenses_sudo = self.env['hr.expense'].sudo().search([
```

Comment:
I assume the sudo() is there for multicompany purposes, but do we really need that ? :thinking: standard_price is a company-dependent field anyway; isn't it ? :thinking: